pkg/client/client_builder.go

```diff
@@ -0,0 +1,153 @@
+// Copyright 2019 Red Hat, Inc. and/or its affiliates
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package client
+
+import (
+	"fmt"
+	buildv1 "github.com/openshift/client-go/build/clientset/versioned/typed/build/v1"
+	imagev1 "github.com/openshift/client-go/image/clientset/versioned/typed/image/v1"
+	"k8s.io/client-go/discovery"
+	restclient "k8s.io/client-go/rest"
+	controllercli "sigs.k8s.io/controller-runtime/pkg/client"
+
+	monclientv1 "github.com/coreos/prometheus-operator/pkg/client/versioned/typed/monitoring/v1"
+)
+
+// Builder wraps information about what to create for a client before building it
+type Builder interface {
+	UseConfig(kubeconfig *restclient.Config) Builder
```

Comment:
The method documentation must be into the interface instead of in the implementation.